src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -1366,6 +1366,72 @@ public void testRestartContainer() throws Exception {
   }
 
   @Test
+  public void testKillContainer() throws Exception {
+    sut.pull(BUSYBOX_LATEST);
+
+    final ContainerConfig containerConfig = ContainerConfig.builder()
+            .image(BUSYBOX_LATEST)
+            // make sure the container's busy doing something upon startup
+            .cmd("sh", "-c", "while :; do sleep 1; done")
+            .build();
+    final String containerName = randomName();
+    final ContainerCreation containerCreation = sut.createContainer(containerConfig, containerName);
+    final String containerId = containerCreation.id();
+
+    sut.startContainer(containerId);
+
+    // Must be running
+    {
+      final ContainerInfo containerInfo = sut.inspectContainer(containerId);
+      assertThat(containerInfo.state().running(), equalTo(true));
+    }
+
+    sut.killContainer(containerId);
+
+    // Should not be running
+    {
+      final ContainerInfo containerInfoLatest = sut.inspectContainer(containerId);
```

Comment:
just curious: why wrap this in a block if you use different names for the ContainerInfo instances and therefore there is no name collision?